src/measure_theory/function/strongly_measurable.lean

```diff
@@ -0,0 +1,328 @@
+  such that for all `n ∈ ℕ`, the measure of the support of `fs n` is finite.
+* `ae_fin_strongly_measurable f μ`: `f` is almost everywhere equal to a `fin_strongly_measurable`
+  function.
+
+* `ae_fin_strongly_measurable.sigma_finite_set`: a measurable set `t` such that
+  `f =ᵐ[μ.restrict tᶜ] 0` and `μ.restrict t` is sigma-finite.
+
+## Main statements
+
+* `ae_fin_strongly_measurable.exists_set_sigma_finite`: there exists a measurable set `t` such that
+  `f =ᵐ[μ.restrict tᶜ] 0` and `μ.restrict t` is sigma-finite.
+* `mem_ℒp.ae_fin_strongly_measurable`: if `mem_ℒp f p μ` with `0 < p < ∞`, then
+  `ae_fin_strongly_measurable f μ`.
+* `Lp.fin_strongly_measurable`: for `0 < p < ∞`, `Lp` functions are finitely strongly measurable.
+
+## References
+
+* Hytönen, Tuomas, Jan Van Neerven, Mark Veraar, and Lutz Weis. Analysis in Banach spaces.
+  Springer, 2016.
+
+-/
+
+open measure_theory filter topological_space function
+open_locale ennreal topological_space measure_theory
+
+namespace measure_theory
+
+local infixr ` →ₛ `:25 := simple_func
+
+section definitions
+variables {α β : Type*} [topological_space β]
+
+/-- A function is `strongly_measurable` if it is the limit of simple functions. -/
+def strongly_measurable [measurable_space α] (f : α → β) : Prop :=
+∃ fs : ℕ → α →ₛ β, ∀ x, tendsto (λ n, fs n x) at_top (𝓝 (f x))
+
+/-- A function is `fin_strongly_measurable` with respect to a measure if it is the limit of simple
+  functions with support with finite measure. -/
+def fin_strongly_measurable [has_zero β] {m0 : measurable_space α} (f : α → β) (μ : measure α) :
+  Prop :=
+∃ fs : ℕ → α →ₛ β, (∀ n, μ (support (fs n)) < ∞) ∧ (∀ x, tendsto (λ n, fs n x) at_top (𝓝 (f x)))
+
+/-- A function is `ae_fin_strongly_measurable` with respect to a measure if it is almost everywhere
+equal to the limit of a sequence of simple functions with support with finite measure. -/
+def ae_fin_strongly_measurable [has_zero β] {m0 : measurable_space α} (f : α → β) (μ : measure α) :
+  Prop :=
+∃ g, fin_strongly_measurable g μ ∧ f =ᵐ[μ] g
+
+end definitions
+
+/-! ## Strongly measurable functions -/
+
+namespace strongly_measurable
+
+variables {α β : Type*} {f : α → β}
+
+/-- A sequence of simple functions such that `∀ x, tendsto (λ n, hf.approx n x) at_top (𝓝 (f x))`.
+That property is given by `strongly_measurable.tendsto_approx`. -/
+protected noncomputable
+def approx [measurable_space α] [topological_space β] (hf : strongly_measurable f) : ℕ → α →ₛ β :=
+hf.some
+
+protected lemma tendsto_approx [measurable_space α] [topological_space β]
+  (hf : strongly_measurable f) :
+  ∀ x, tendsto (λ n, hf.approx n x) at_top (𝓝 (f x)) :=
+hf.some_spec
+
+lemma fin_strongly_measurable_of_exists_set_sigma_finite [topological_space β] [has_zero β]
+  {m : measurable_space α} {μ : measure α} (hf_meas : strongly_measurable f)
+  (hf : ∃ t, measurable_set t ∧ (∀ x ∈ tᶜ, f x = 0) ∧ sigma_finite (μ.restrict t)) :
```

Comment:
when you have an existential assumption like that, it is simpler to take `t` as an argument, and the three conditions separately on `t`.